CountingSort/app_resources/scatter_shader.comp.hlsl

```diff
@@ -10,7 +11,13 @@ uint32_t3 nbl::hlsl::glsl::gl_WorkGroupSize()
 [numthreads(WorkgroupSize, 1, 1)]
 void main(uint32_t3 ID : SV_GroupThreadID, uint32_t3 GroupID : SV_GroupID)
 {
-    nbl::hlsl::sort::counting < bda::__ptr < uint32_t >, bda::__ptr < uint32_t >, bda::__ptr < uint32_t >  > counter;
-    counter.init(pushData);
-    counter.scatter();
+    nbl::hlsl::sort::counting < bda::PtrAccessor<uint32_t>, bda::PtrAccessor<uint32_t>, bda::PtrAccessor<uint32_t> > counter;
+    counter.scatter(
+        bda::PtrAccessor<uint32_t>::createAccessor(pushData.inputKeyAddress),
+        bda::PtrAccessor<uint32_t>::createAccessor(pushData.inputValueAddress),
+        bda::PtrAccessor<uint32_t>::createAccessor(pushData.scratchAddress),
+        bda::PtrAccessor<uint32_t>::createAccessor(pushData.outputKeyAddress),
+        bda::PtrAccessor<uint32_t>::createAccessor(pushData.outputValueAddress),
```

Comment:
I'd roll up the input/output accessor pairs into one, because they're accessors you can make a custom one that gets from a different address than sets

Reply:
Done